lib/mysql_connection.cpp

```diff
@@ -2569,6 +2572,34 @@ void MySQL_Connection::ProcessQueryAndSetStatusFlags(char *query_digest_text) {
 			}
 		}
 	}
+	// checking warnings and disabling multiplexing will be effective only when the mysql-query_digests is enabled
+	if (myds->sess->CurrentQuery.QueryParserArgs.digest_text) {
+		if (get_status(STATUS_MYSQL_CONNECTION_HAS_WARNINGS) == false) {
+			if (mysql_warning_count(this->mysql) > 0) {
```

Comment:
Because calling `mysql_warning_count()` we need to ensure that `this->mysql` is not `NULL` .

Reply:
Done